Libraries/Text/TextInput/RCTBackedTextInputViewProtocol.h

```diff
@@ -59,6 +59,9 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, assign, readonly) CGFloat zoomScale;
 @property (nonatomic, assign, readonly) CGPoint contentOffset;
 @property (nonatomic, assign, readonly) UIEdgeInsets contentInset;
+#if TARGET_OS_OSX // [macOS
+@property (nonatomic, assign) CGFloat pointScaleFactor;
+#endif // macOS]
```

Comment:
I'm not sure I like the idea of adding pointScaleFactor as a property, feels like a mixing of responsibilities. Is there a way to use a default value of 1.0, and/or update it when the view _is_ inserted in a window?

Also in general, how much of Fabric depends on creating views before they're in a window? That seems to be a consistent issue we run into. Is this not an issue on iOS because over there, a scale factor of 0.0 uses the device scale factor? Is the whole "create views before they're inserted" a necessary part of the new renderer?

Reply:
> I'm not sure I like the idea of adding pointScaleFactor as a property, feels like a mixing of responsibilities. Is there a way to use a default value of 1.0, and/or update it when the view is inserted in a window?

The change goes in the way of letting react control the changes in scale factors, because the shadow nodes already have that information. This will then allow us to use react's shadow node tree mutations to update the views when the scale has changed.

> I'm not sure I like the idea of adding pointScaleFactor as a property, feels like a mixing of responsibilities. Is there a way to use a default value of 1.0, and/or update it when the view is inserted in a window?

Fabric builds partial host view hierarchies before inserting them in a parent hierarchy. Enforcing a specific order for the creation of the host view tree is a big change.

The other way to fix this would be to register the views to listen to `- (void)viewDidChangeBackingProperties` and re-render outside of react's control, but we're trying to move away from out-of-react direct view manipulation so that we can support any future optimizations by giving as much control to the RN renderer.

On iOS the assumption is still that there is only one screen at all times. At some point, RN will have to be updated to support multiple screens because of iPadOS supporting external displays. And from the discussions I've had, it would be handled through the shadow tree to avoid direct view manipulations.

Reply:
Could you talk a bit more about how iOS would handle external screen support in the shadow tree? My current feeling is that the "correct" way to implement this would be listening to `viewDidChangeBackingProperties` (we do that a bit in Paper I think?), but I understand that's taking control out of the RN renderer. 

If we went this way for TextInput, would we need to add this for other components as well? Would that then be a macOS specific difference that needs to be accounted in say, 3rd party community modules as well?

Reply:
In the fabric ComponentView implementations, the scale factor is provided through the `layoutMetrics`. It looks odd for the TextInput because we're re-using the paper backing views.

But from the discussions I've had, the root view would listen for changes in backing properties. Changes in scale would then trigger a clone of the shadow tree that would be committed, which would notify the host views of the changes. The host views would then decide if they need to act on the change or not, just like a state change.

Reply:
Your last response has me more open to the change. I only haven't merged as we haven't had a chance to talk about more generally yet.

I imagine dpi changes is something only RN desktop has to worry about and not RN core anytime soon? So this is probably an area where we're unique =)

Reply:
It will be needed on RN core, but it is a rare use-case.

Happy to discuss it further. I'll also submit a PR that does add support for a RN-driven refresh of the components in fabric when they move to a display with a different scale factor. This will add more context to this change.